nipype/pipeline/plugins/multiproc.py

```diff
@@ -53,50 +87,223 @@ class MultiProcPlugin(DistributedPluginBase):
 
-    def _clear_task(self, taskid):
-        del self._taskresult[taskid]
+    def _send_procs_to_workers(self, jobids, updatehash=False, graph=None):
+        """
+        Sends jobs to workers
+        """
+        jobids = np.atleast_1d(jobids).tolist()
+        forkjids = []
+        num_jobs = len(jobids)
+        if np.isinf(self.max_jobs):
+            slots = None
+        else:
+            slots = max(0, self.max_jobs - num_jobs)
+
+        if num_jobs > 0:
+            # send all available jobs
+            LOGGER.info('Check %d jobs and submit' % len(jobids[:slots]))
+            for jobid in jobids[:slots]:
+                if isinstance(self.procs[jobid], MapNode):
+                    try:
+                        num_subnodes = self.procs[jobid].num_subnodes()
+                    except Exception:
+                        self._clean_queue(jobid, graph)
+                        self.proc_pending[jobid] = False
+                        continue
+                    if num_subnodes > 1:
+                        submit = self._submit_mapnode(jobid)
+                        if not submit:
+                            continue
+                # change job status in appropriate queues
+                self.proc_pending[jobid] = True
+                # Send job to task manager and add to pending tasks
+                if self._status_callback:
+                    self._status_callback(self.procs[jobid], 'start')
+                continue_with_submission = True
+                if str2bool(self.procs[jobid].config['execution']
+                            ['local_hash_check']):
+                    LOGGER.info('checking hash locally')
+                    hash_exists, _, _, _ = self.procs[jobid].hash_exists()
+                    overwrite = getattr(
+                        self.procs[jobid], 'overwrite', False)
+                    always_run = getattr(
+                        self.procs[jobid]._interface, 'always_run', False)
+
+                    # Is cached and run enforced
+                    if (hash_exists and not overwrite and not always_run):
+                        continue_with_submission = False
+                        self.proc_done[jobid] = True
+                        self._task_finished_cb(jobid)
+                        self._remove_node_dirs()
+                        LOGGER.info(('Node %s (%d) is cached or does not require being run') %
+                                    (self.procs[jobid], jobid))
+
+                if continue_with_submission:
+                    self.pending_tasks.insert(0, jobid)
+                    sworker = getattr(self.procs[jobid]._interface,
+                                      '_singleworker', True)
+                    nosubmit = getattr(self.procs[jobid],
+                                       'run_without_submitting', False)
+                    if sworker and not nosubmit:
+                        forkjids.append(jobid)
+                    else:
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Too many positional arguments for function call](https://www.codacy.com/app/krzysztof-gorgolewski/nipy_nipype/file/1940156842/issues/source?bid=2767207&fileBranchId=2790691#l261)